Lib/test/test_os.py

```diff
@@ -1419,9 +1419,20 @@ def test_urandom_failure(self):
             import errno
             import os
             import resource
-
             soft_limit, hard_limit = resource.getrlimit(resource.RLIMIT_NOFILE)
-            resource.setrlimit(resource.RLIMIT_NOFILE, (1, hard_limit))
+            try:
+                resource.setrlimit(resource.RLIMIT_NOFILE, (1, hard_limit))
+            except (ValueError,OSError):
+                for i in range(1, hard_limit + 1):
+                    try:
+                        os.open(os.devnull, os.O_RDONLY)
```

Comment:
I dislike this code. If the limit is high (ex: larger than 1024), the test can take a lot of time.

I suggest to simply skip the test on VxWorks.

Reply:
Following your comments, skip this test on VxWorks. By default the RLIMIT_NOFILE on VxWorks is 8. But on other OS, perhaps there will be the issues you concerned.